src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -0,0 +1,146 @@
+package com.pusher.client.channel.impl;
+
+import com.google.gson.Gson;
+import com.pusher.client.AuthorizationFailureException;
+import com.pusher.client.Authorizer;
+import com.pusher.client.channel.PrivateEncryptedChannel;
+import com.pusher.client.channel.PrivateEncryptedChannelEventListener;
+import com.pusher.client.channel.SubscriptionEventListener;
+import com.pusher.client.connection.impl.InternalConnection;
+import com.pusher.client.crypto.nacl.SecretBoxOpener;
+import com.pusher.client.util.Base64;
+import com.pusher.client.util.Factory;
+
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+public class PrivateEncryptedChannelImpl extends ChannelImpl implements PrivateEncryptedChannel {
+
+    private static final Gson GSON = new Gson();
+    private final InternalConnection connection;
+    private final Authorizer authorizer;
+    private SecretBoxOpener secretBoxOpener;
+
+    private class PrivateEncryptedChannelData {
+        byte[] auth;
+        final String channelData;
+
+        protected PrivateEncryptedChannelData(String auth, String channelData) {
```

Comment:
I created this class to contain the things the private encrypted channel needs. 

I created a `prepare()` method to parse the json into this object once, and do all the checks to make sure we got all the things we needed.

I also created a `tearDown()` method to clear all the private data more securely.